packages/cli/src/collect/collect.js

```diff
@@ -174,7 +178,21 @@ async function startServerAndDetermineUrls(options) {
   const urls = urlsAsArray;
   if (!urls.length) {
     const maxNumberOfUrls = options.maxAutodiscoverUrls || Infinity;
-    urls.push(...server.getAvailableUrls().slice(0, maxNumberOfUrls));
+    const autodiscoverUrlBlocklistAsArray = Array.isArray(options.autodiscoverUrlBlocklist)
+      ? options.autodiscoverUrlBlocklist
+      : options.autodiscoverUrlBlocklist
+      ? [options.autodiscoverUrlBlocklist]
+      : [];
+    const availableUrls = server.getAvailableUrls();
+    const normalizedBlocklist = autodiscoverUrlBlocklistAsArray.map(
+      rawUrl => {
+        const url = new URL(rawUrl, 'http://localhost');
+        url.port = server.port.toString();
+        return url.href;
+      }
+    );
+    const urlsToUse = availableUrls.filter(url => !normalizedBlocklist.includes(url)).slice(0, maxNumberOfUrls)
```

Comment:
```suggestion
    const urlsToUse = availableUrls.filter(url => !normalizedBlocklist.includes(url)).slice(0, maxNumberOfUrls);
```

I suspect this line might fail lint FYI